wagtail/admin/templates/wagtailadmin/shared/header.html

```diff
@@ -14,15 +14,19 @@
     - `action_icon` - icon for the 'action' button, default is 'icon-plus'
 
 {% endcomment %}
-<header class="header {% if merged %}merged{% endif %} {% if search_form %}hasform{% endif %}">
+<header class="header{% if merged %} merged{% endif %}{% if search_form %}hasform{% endif %} {{ classname }}">
     {% block breadcrumb %}{% endblock %}
     <div class="row">
         <div class="left">
             <div class="col">
                 <h1 class="header__title">
-                    {% if icon %}{% icon name=icon %}{% endif %}
+                    {% if icon %}
+                        {% icon class_name="w-header__icon" name=icon %}
+                    {% elif avatar %}
+                        <div class="avatar"><img src="{% avatar_url user %}" alt="" /></div>{% endif %}
```

Comment:
Be careful with the way you are reading the avatar here, you are reading it from context but not passing it in. instead the idea is that the `avatar` variable is the fully resolved URL.

`src="{{ avatar }}}`

Otherwise this template becomes more intertwined with requiring both the user and the template tag `avatar_url` in the context (this may also break storybook).